addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5326,20 +5302,93 @@ registry.layout_column = SnippetOptionWidget.extend({
         }
         // Removing the grid properties.
         delete rowEl.dataset.rowCount;
-        rowEl.style.removeProperty('--grid-item-padding-x');
+        rowEl.style.removeProperty('--grid-item-padding-x'); // Kept for compatibility.
         rowEl.style.removeProperty('--grid-item-padding-y');
     },
+    /**
+     * Removes all the eventual grid previews, to avoid unwanted behaviors (when
+     * toggling normal mode for example).
+     *
+     * @private
+     */
+    _clearAllPreviews(columnEls) {
+        // Removing the padding previews.
+        for (const columnEl of columnEls) {
+            if (columnEl._removePaddingPreview) {
+                columnEl._removePaddingPreview();
+            }
+        }
+    },
+});
+
+registry.GridColumns = SnippetOptionWidget.extend({
+    /**
+     * @override
+     */
+    cleanForSave() {
+        // Remove the padding highlights.
+        if (this.$target[0].classList.contains("o_we_padding_highlight")) {
+            this._removePaddingPreview();
+        }
+    },
+    /**
+     * @override
+     */
+    cleanUI() {
+        // Remove the padding highlights.
+        if (this.$target[0].classList.contains("o_we_padding_highlight")) {
+            this._removePaddingPreview();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async selectStyle(previewMode, widgetValue, params) {
+        await this._super(...arguments);
+        if (params.cssProperty.startsWith("--grid-item-padding")) {
```

Comment:
I'd prefer you'd list them both instead of relying on `startsWith`. Would be easier not to make a mistake while grepping.

Reply:
Indeed, forgot to change it when changing the other :grimacing: